selenium/src/main/scala/not/ogame/bots/selenium/MyFleetsComponentReader.scala

```diff
@@ -79,4 +80,45 @@ class MyFleetsComponentReader(webDriver: WebDriver)(implicit clock: LocalClock)
     val missionTextLocalized = fleetElement.findElement(By.className("mission")).getText
     missionTextLocalized.contains("(R)")
   }
+
+  private def getShips(fleetElement: WebElement): Map[ShipType, Int] = {
+    val fleetInfoTooltip = fleetElement.findElement(By.className("fleetinfo"))
+    val rows = fleetInfoTooltip.findElementsS(By.tagName("tr"))
+    val value: List[Option[(ShipType, Int)]] = rows.map(row => {
+      val rowText = row.getAttribute("textContent")
+      getShipType(rowText).map(shipType => {
+        val count = row.findElement(By.className("value")).getAttribute("textContent").filter(_.isDigit).toInt
+        (shipType, count)
+      })
+    })
+    value.filter(option => option.isDefined).map(option => option.get).toMap.withDefaultValue(0)
+  }
+
+  private def getShipType(rowText: String): Option[ShipType] = {
+    nameOfShips().find { case (_, shipName) => rowText.contains(shipName) }.map { case (shipType, _) => shipType }
```

Comment:
I think that you can leverage power of `collectFirst` combined with pattern matching here ;)

Reply:
I think that I can not :)

Reply:
Don't get me wrong I tried and after initial panic when seeing PartialFunction interface I only managed to replace `find` with `collectFirst` and noting else changed.

Reply:
```
nameOfShips().collectFirst { case (shipType, shipName) if rowText.contains(shipName) => shipType }
```